### PR TITLE
feat(init): Initial commit

### PR DESCRIPTION
**Pull request by n0th1ng-else**
- Add package file
- Setup ESLint
- Setup Prettier
- Setup Typescript
- Setup Jest
- Initial store logic covered by tests
- setup circle-ci config
- setup circe-ci github reporter
- setup Jest to make coverage and test reports
- setup semantic-release

### CODE REVIEW

**Comment by n0th1ng-else** [comment]
<h3>chrome(circleci): Setup circle ci bot</h3>
Demo: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/18/artifacts/0//home/circleci/project/artifacts/index.html' target='_blank'>demo</a></strong>

**Comment by n0th1ng-else** [comment]
<h3>chrome(circleci): Setup circle ci bot</h3>
Demo: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/20/artifacts/0//home/circleci/project/artifacts/index.html' target='_blank'>demo</a></strong>

**Comment by n0th1ng-else** [comment]
<h3>chrome(circleci): Setup circle ci bot</h3>
<div>Coverage: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/22/artifacts/0//home/circleci/project/coverage/index.html' target='_blank'>report</a></strong></div>
<div>Test results: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/22/artifacts/0//home/circleci/project/test-report/index.html' target='_blank'>report</a></strong></div>

**Comment by n0th1ng-else** [comment]
<h3>chrome(circleci): Setup circle-ci - setup circle-ci config - setup circe-ci github reporter - setup Jest to make coverage and test reports</h3>
<div>Coverage: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/24/artifacts/0//home/circleci/project/coverage/index.html' target='_blank'>report</a></strong></div>
<div>Test results: <strong><a href='https://circleci.com/gh/n0th1ng-else/micro-store/24/artifacts/0//home/circleci/project/test-report/index.html' target='_blank'>report</a></strong></div>

File: src/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+		});
+
+		describe('should update', () => {
+			it('store with any apppropriate value', done => {
+				const store = writableStore<string>();
+
+				const handler = jest
+					.fn()
+					.mockImplementationOnce(data => {
+						expect(data).toBeNull();
```
Comment (irinaklimova) [inline src/index.spec.ts:30]: Try to stick to one style: `toBeNull()` or `toBe(null)` (line 9)

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+function notify<D>(item: WeakData<D>, subscriptions: Array<DataSubscription<D>>): void {
+	subscriptions.forEach(handlers => {
+		handlers.subscription(item);
+	});
+}
+
+function subscribeStore<D>(
+	item: WeakData<D>,
+	subscribers: Array<DataSubscription<D>>,
```
Comment (irinaklimova) [inline src/index.ts]: Do you mean `subscriptions`?

File: src/index.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		unsubscribe(subscription: DataSubscriber<D>): void {
+			subscribers = unsubscribeStore(subscribers, subscription);
+		},
+		unsubscribeAll(): void {
+			subscribers = unsubscribeStore(subscribers);
+		}
+	};
+};
+
+export const freezableStore = <D>(item: WeakData<D> = null, freezeCount = 1) => {
```
Comment (irinaklimova) [inline src/index.ts:23]: Can you split this file at least on two?

**Review by irinaklimova — APPROVED** [review]

**Pull request merged**